### PR TITLE
[17.0] [MIG] account_payment_partner

### PR DESCRIPTION
**Pull request by miquelalzanillas**
Depends on:

- [#1182]

### CODE REVIEW

**Comment by miquelalzanillas** [comment]
Hi @pedrobaeza , tests are failing for a missing module dependency for which I also have an open PR [#1182](https://github.com/OCA/bank-payment/pull/1182). Do I have to wait now until the PR is merged for push again here to re-run tests?

**Comment by pedrobaeza** [comment]
https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference%28s%29-to-another-pull-request%28s%29

**Comment by miquelalzanillas** [comment]
Thanks!

**Comment by miquelalzanillas** [comment]
Ready. Please review: @peluko00 @lbarry-apsl @javierobcn

**Review by samoros1 — APPROVED** [review]

**Review by lbarry-apsl — APPROVED** [review]
Validated functionality

File: account_payment_partner/views/account_move_line.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='analytic_distribution']/.." position="after">
+                <group name="payments" string="Payments">
+                    <field name="account_type" invisible="1" />
+                    <field name="reconciled" invisible="1" />
+                    <field
+                        name="payment_mode_id"
+                        force_save="1"
+                        invisible="account_type not in ['asset_receivable', 'liability_payable']"
+                        readonly="account_type not in ['asset_receivable', 'liability_payable'] or reconciled == True"
```
Comment (pedrobaeza) [inline account_payment_partner/views/account_move_line.xml]: And the same for similar cases with booleans:

```suggestion
                        readonly="account_type not in ['asset_receivable', 'liability_payable'] or reconciled"
```

Comment (miquelalzanillas) [reply account_payment_partner/views/account_move_line.xml]: Fixed. Thanks.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

File: account_payment_partner/models/account_move.py [hunk 2]
```diff
@@ -90,12 +90,29 @@ def _compute_payment_mode_id(self):
...
+                            and payment_mode.fixed_journal_id.bank_account_id
+                        ):
+                            move.partner_bank_id = (
+                                payment_mode.fixed_journal_id.bank_account_id
+                            )
+                            continue
+                    else:
+                        move.partner_bank_id = False
+            else:
                 move.partner_bank_id = False
```
Comment (rousseldenis) [inline account_payment_partner/models/account_move.py:115]: @miquelalzanillas @grindtildeath I'm wondering why removing the bank if it is already set and no payment mode ?

**Pull request merged**